DQM/HcalTasks/plugins/HcalGPUComparisonTask.cc

```diff
@@ -0,0 +1,180 @@
+// -*- C++ -*-
+//
+
+#include "DQM/HcalCommon/interface/DQTask.h"
+#include "DQM/HcalCommon/interface/Utilities.h"
+#include "DQM/HcalCommon/interface/HashFilter.h"
+#include "DQM/HcalCommon/interface/Container1D.h"
+#include "DQM/HcalCommon/interface/Container2D.h"
+#include "DQM/HcalCommon/interface/ContainerProf1D.h"
+#include "DQM/HcalCommon/interface/ContainerProf2D.h"
+#include "DQM/HcalCommon/interface/ContainerSingle1D.h"
+#include "DQM/HcalCommon/interface/ContainerSingle2D.h"
+#include "DQM/HcalCommon/interface/ContainerSingleProf2D.h"
+#include "DQM/HcalCommon/interface/ElectronicsMap.h"
+
+#include <cmath>
+
+using namespace hcaldqm;
+using namespace hcaldqm::constants;
+using namespace hcaldqm::filter;
+
+class HcalGPUComparisonTask : public hcaldqm::DQTask {
+public:
+  HcalGPUComparisonTask(edm::ParameterSet const&);
+  ~HcalGPUComparisonTask() override {}
+
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  std::shared_ptr<hcaldqm::Cache> globalBeginLuminosityBlock(edm::LuminosityBlock const&,
+                                                             edm::EventSetup const&) const override;
+  void globalEndLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void _process(edm::Event const&, edm::EventSetup const&) override;
+  void _resetMonitors(hcaldqm::UpdateFreq) override;
+
+  edm::InputTag tagHBHE_ref_;     //CPU version
+  edm::InputTag tagHBHE_target_;  //GPU version
+  edm::EDGetTokenT<HBHERecHitCollection> tokHBHE_ref_;
+  edm::EDGetTokenT<HBHERecHitCollection> tokHBHE_target_;
+  edm::ESGetToken<HcalDbService, HcalDbRecord> hcalDbServiceToken_;
+
+  //    GPU reco test verification
+  hcaldqm::Container2D energyGPUvsCPU_subdet_;
+  hcaldqm::Container1D energyDiffGPUCPU_subdet_;
+  hcaldqm::Container2D energyDiffGPUCPU_depth_;
+};
+
+HcalGPUComparisonTask::HcalGPUComparisonTask(edm::ParameterSet const& ps)
+    : DQTask(ps), hcalDbServiceToken_(esConsumes<HcalDbService, HcalDbRecord, edm::Transition::BeginRun>()) {
+  tagHBHE_ref_ = ps.getUntrackedParameter<edm::InputTag>("tagHBHE_ref", edm::InputTag("hltHbhereco@cpu"));
+  tagHBHE_target_ = ps.getUntrackedParameter<edm::InputTag>("tagHBHE_target", edm::InputTag("hltHbhereco@cuda"));
```

Comment:
Since DQM is not writing EDM data where provenance can be an issue, we have never worried about untracked/tracked parameters. I'd really thank any advice on this front so that we could advise for upcoming new modules and standarize the existing ones. @makortel  
Thanks